catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/ReindexCommand.java

```diff
@@ -0,0 +1,496 @@
+  }
+
+  @VisibleForTesting
+  protected void setMetacardClient(SolrMetacardClientImpl metacardClient) {
+    this.metacardClient = metacardClient;
+  }
+
+  @VisibleForTesting
+  protected void setNumThread(int numThreads) {
+    this.numThreads = numThreads;
+  }
+
+  @VisibleForTesting
+  protected void setSolrHost(String solrHost) {
+    this.solrHost = solrHost;
+  }
+
+  @VisibleForTesting
+  protected void setCollection(String collection) {
+    this.collection = collection;
+  }
+
+  @VisibleForTesting
+  protected void setSecurity(Security security) {
+    this.security = security;
+  }
+
+  @VisibleForTesting
+  protected void setCatalogFramework(CatalogFramework catalogFramework) {
+    this.catalogFramework = catalogFramework;
+  }
+
+  @VisibleForTesting
+  protected void setAfterDate(String afterDate) {
+    this.afterDate = afterDate;
+  }
+
+  @VisibleForTesting
+  protected void setBeforeDate(String beforeDate) {
+    this.beforeDate = beforeDate;
+  }
+
+  @VisibleForTesting
+  protected void setField(String field) {
+    this.field = field;
+  }
+
+  private boolean isSolrClientAvailable(org.codice.solr.client.solrj.SolrClient solrClient) {
+    RetryPolicy retryPolicy =
+        new RetryPolicy()
+            .withDelay(100, TimeUnit.MILLISECONDS)
+            .withMaxDuration(3, TimeUnit.MINUTES)
+            .retryWhen(false);
+    Failsafe.with(retryPolicy).get((Callable<Boolean>) solrClient::isAvailable);
+    return solrClient.isAvailable();
+  }
+
+  private void migrate(org.codice.solr.client.solrj.SolrClient sourceSolr) {
+    startDataQueryThread(sourceSolr);
+    startDataWriterThreads();
+  }
+
+  private void startDataWriterThreads() {
+    ThreadFactory threadFactory = Executors.defaultThreadFactory();
+    publishExecutor =
+        new ThreadPoolExecutor(
+            numThreads,
+            numThreads,
+            0L,
+            TimeUnit.MILLISECONDS,
+            new LinkedBlockingDeque<>(numThreads),
+            threadFactory,
+            new ThreadPoolExecutor.CallerRunsPolicy());
+  }
+
+  private void startDataQueryThread(org.codice.solr.client.solrj.SolrClient sourceSolr) {
+    readerThread = new Reader(sourceSolr);
+    readerThread.setName("reindex-reader");
+    readerThread.start();
+  }
+
+  private void stopWorkers(boolean force) {
+    if (readerThread != null && readerThread.isRunning()) {
+      readerThread.interrupt();
+    }
+
+    if (publishExecutor != null) {
+      if (!publishExecutor.isShutdown()) {
+        if (force) {
+          publishExecutor.shutdownNow();
+        } else {
+          publishExecutor.shutdown();
+        }
+      }
+    }
+  }
+
+  private void waitForCompletion() throws InterruptedException {
+    boolean running = true;
+    if (readerThread == null) {
+      return;
+    }
+
+    while (running) {
+      if (!readerThread.isRunning()) {
+        publishExecutor.shutdown();
+        publishExecutor.awaitTermination(15, TimeUnit.MINUTES);
```

Comment:
that was just conservative worst case scenario guess for the time to ingest 200 records. Ingest could be expensive if the record is large, with many preingest plugins and transformers are active. But it could definitely be lower.